nonce/nonce.go

```diff
@@ -281,6 +289,7 @@ func (ns *NonceService) Valid(nonce string) bool {
 	if len(ns.used) > ns.maxUsed {
 		s := time.Now()
 		ns.earliest = heap.Pop(ns.usedHeap).(int64)
+		ns.nonceEarliest.Set(float64(ns.earliest))
```

Comment:
If you wanted to do this as a counter to match nonce_redeems, I think this could become
```suggestion
		prev := ns.earliest
		ns.earliest = heap.Pop(ns.usedHeap).(int64)
		ns.nonceEarliest.Add(ns.earliest - prev)
```

But frankly I think that a gauge is conceptually more appropriate for this anyway, as long as that won't cause problems for querying/comparing both this and nonce_redeems in the same grafana graph.

Reply:
I considered that approach but it seems more likely to somehow get out of sync or otherwise be error prone (eg, if we add another codepath that changed earliest) without much gain.  I don't think there's going to be any problems introduced by it being a gauge.